packages/cockpit/ui/src/components/Contracts.js

```diff
@@ -17,23 +17,22 @@ const Contracts = ({contracts, changePage, currentPage, numberOfPages, title = "
         <CardBody>
           {!contracts.length && "No contracts to display"}
           {contracts
-            .map((contract, key) => {
-              const contractDisplay = formatContractForDisplay(contract);
-              if (!contractDisplay) {
-                return '';
+           .map((contract, key) => {
+              if (!contract) {
+                return null;
```

Comment:
I'm sure there exists a good reason for this, but I find it unexpected that `contract` can be `null` here. `contracts` should IMO only hold valid objects so that the view doesn't have to worry about this kind of logic and simply render data.

Do you think we can maybe put a `filter(contract => !!contract)` on `contracts` inside the component before we iterate over it here?

Reply:
I agree that it's weird that a contract in the array would be falsey, but in the end doing the check here is not the end of the world, because using `filter` would be an additional loop that is not 100% needed

Reply:
Prior to the changes in this PRm the null check took place [after the call](https://github.com/embark-framework/embark/blob/master/packages/cockpit/ui/src/components/Contracts.js#L22-L24) to [`formatContractForDisplay`](https://github.com/embark-framework/embark/blob/master/packages/cockpit/ui/src/utils/presentation.js#L4). However, that function always returns an object, so the null check didn't make sense; also `formatContractForDisplay` isn't written to handle possibly falsy values. To be on the safe side, I simply moved the check up a bit, but I wasn't and am not sure if it's really possible that the contracts array might contain falsy values. It's definitely something we can investigate/refactor further, and I agree that `contracts` should hold only valid objects.